### PR TITLE
Setup Python class to represent name db

### LINKED ISSUE #1214: Custom python class to represent `"char"` DB type
## Problem

Currently, a columns of types `NAME` or `"char"` in the database show up as type `VARCHAR` in the API. This is because SQLAlchemy reflects these types as the `String` python class, which compiles to `VARCHAR` in the `postgres` dialect.

## Proposed solution

We should:
- create a `UserDefinedType` representing `"char"`, and another representing `NAME`.
- These types should be mapped to by the `engine.dialect.ischema_names` dictionary under the `'"char"'` and `'name'` keys, respectively. (note that the string contains double-quotes)

## Additional context

Based on a discussion: #1184

### PR DESCRIPTION
**Pull request by Cronus1007**
Fixes #1214 

**Technical details**
Initial setup of the python class to represent 'name' db Type
**Screenshots**

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: db/types/base.py [hunk 1]
```diff
@@ -46,6 +46,7 @@ class PostgresType(Enum):
     JSONB = 'jsonb'
     MACADDR = 'macaddr'
     MATHESAR_CHAR = '"char"'
+    MATHESAR_NAME = 'name'
     MONEY = 'money'
     NAME = 'name'
```
Comment (mathemancer) [inline db/types/base.py:51]: Only one of `MATHESAR_NAME` or `NAME` should remain.

File: db/types/operations/cast.py [hunk 2]
```diff
@@ -40,6 +40,7 @@
 FULL_CHAR = base.PostgresType.CHARACTER.value
 NAME = base.PostgresType.NAME.value
 MATHESAR_CHAR = base.PostgresType.MATHESAR_CHAR.value
+MATHESAR_NAME = base.PostgresType.MATHESAR_NAME.value
```
Comment (mathemancer) [inline db/types/operations/cast.py:43]: Same here. We shouldn't have both. We want to replace the default `"name"` implementation with ours (mainly in the `ischema_names` dict).

**Review by mathemancer — CHANGES_REQUESTED** [review]
Thanks for your work on this so far. 

We shouldn't keep two versions of `NAME` around (i.e., `NAME` and `MATHESAR_NAME`. The goal is to replace the default SQLAlchemy implementation of the `NAME` type with ours.

**Comment by Cronus1007** [comment]
@mathemancer Closing #1387 for now since #1222 introduced a lot of merge conflicts so I would like to have a fresh start with the changes u suggested.

**Comment by mathemancer** [comment]
@Cronus1007 Sorry for the confusion on this. Please let us know if there's any problem trying to reorganize with the new structure.

**Comment by Cronus1007** [comment]
@mathemancer Sure.

**Pull request closed without merging**